ode/impl/KokkosODE_RungeKutta_impl.hpp

```diff
@@ -97,20 +97,38 @@ KOKKOS_FUNCTION Experimental::ode_solver_status RKSolve(
     const vec_type& y, const vec_type& temp, const mv_type& k_vecs) {
   constexpr scalar_type error_threshold = 1;
   bool adapt                            = params.adaptivity;
+  bool dt_was_reduced;
   if (std::is_same_v<table_type, ButcherTableau<0, 0>>) {
     adapt = false;
   }
 
-  scalar_type dt = (t_end - t_start) / params.max_steps;
-  scalar_type t  = t_start;
-  for (int stepIdx = 0; (stepIdx < params.max_steps) && (t < t_end);
+  // Set current time and initial time step
+  scalar_type t_now = t_start;
+  scalar_type dt    = (t_end - t_start) / params.max_steps;
```

Comment:
If scalar_type amd max_steps are integers, is integer truncation safe here?

Reply:
Yep, although t_end and t_start are floating point values.